lib/descriptor.js

```diff
@@ -24,3 +26,67 @@ const generateDescriptor = async (data, dpName = "scratchpad") => {
   return descriptor
 }
 module.exports.generateDescriptor = generateDescriptor
+
+const generateDescriptorForLocalFiles = function (dpName = "scratchpad") { 
+  return new Promise(function(resolve, reject) { 
```

Comment:
I'm not sure about returning Promise here? Is there reason for doing this? 
@anuveyatsu can you take a look at this, as I don't really understand what's going on here